### PR TITLE
fix asan UT fail

### LINKED ISSUE #5600: tiflash-sanitizer-daily fail
## Bug Report

Please answer these questions before submitting your issue. Thanks!

https://ci.pingcap.net/blue/organizations/jenkins/tiflash-sanitizer-daily/detail/tiflash-sanitizer-daily/440/pipeline/

```
[2022-08-10T19:22:45.950Z] FAILED TESTS (2/1454):
[2022-08-10T19:22:45.950Z]     1393 ms: /tiflash/gtests_dbms PlannerInterpreterExecuteTest.ParallelQuery
[2022-08-10T19:22:45.950Z]     1292 ms: /tiflash/gtests_dbms PlannerInterpreterExecuteTest.Join
```
```
[2022-08-10T19:18:02.797Z] ==41802==ERROR: AddressSanitizer: stack-use-after-scope on address 0x7fffff74a461 at pc 0x000003877d76 bp 0x7fffff74a0c0 sp 0x7fffff749868
[2022-08-10T19:18:02.797Z] READ of size 12 at 0x7fffff74a461 thread T0
[2022-08-10T19:18:02.797Z]     #0 0x3877d75 in MemcmpInterceptorCommon(void*, int (*)(void const*, void const*, unsigned long), void const*, void const*, unsigned long) /llvm-project/compiler-rt/lib/asan/../sanitizer_common/sanitizer_common_interceptors.inc:860:7
[2022-08-10T19:18:02.797Z]     #1 0x38782c9 in bcmp /llvm-project/compiler-rt/lib/asan/../sanitizer_common/sanitizer_common_interceptors.inc:906:10
[2022-08-10T19:18:02.797Z]     #2 0x1adf5181 in std::__1::char_traits<char>::compare(char const*, char const*, unsigned long) /usr/local/bin/../include/c++/v1/__string:383:12
[2022-08-10T19:18:02.797Z]     #3 0x1adf5181 in std::__1::basic_string_view<char, std::__1::char_traits<char> >::compare(std::__1::basic_string_view<char, std::__1::char_traits<char> >) const /usr/local/bin/../include/c++/v1/string_view:397:24
[2022-08-10T19:18:02.797Z]     #4 0x1adf5181 in bool std::__1::operator==<char, std::__1::char_traits<char> >(std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> >) /usr/local/bin/../include/c++/v1/string_view:681:18
[2022-08-10T19:18:02.797Z]     #5 0x1adf5181 in std::__1::equal_to<std::__1::basic_string_view<char, std::__1::char_traits<char> > >::operator()(std::__1::basic_string_view<char, std::__1::char_traits<char> > const&, std::__1::basic_string_view<char, std::__1::char_traits<char> > const&) const /usr/local/bin/../include/c++/v1/__functional/operations.h:413:21
[2022-08-10T19:18:02.797Z]     #6 0x1adf5181 in std::__1::pair<std::__1::__hash_iterator<std::__1::__hash_node<std::__1::basic_string_view<char, std::__1::char_traits<char> >, void*>*>, bool> std::__1::__hash_table<std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::hash<std::__1::basic_string_view<char, std::__1::char_traits<char> > >, std::__1::equal_to<std::__1::basic_string_view<char, std::__1::char_traits<char> > >, std::__1::allocator<std::__1::basic_string_view<char, std::__1::char_traits<char> > > >::__emplace_unique_key_args<std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> > >(std::__1::basic_string_view<char, std::__1::char_traits<char> > const&, std::__1::basic_string_view<char, std::__1::char_traits<char> >&&) /usr/local/bin/../include/c++/v1/__hash_table:2087:21
[2022-08-10T19:18:02.797Z]     #7 0x1adf3ad9 in std::__1::__hash_table<std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::hash<std::__1::basic_string_view<char, std::__1::char_traits<char> > >, std::__1::equal_to<std::__1::basic_string_view<char, std::__1::char_traits<char> > >, std::__1::allocator<std::__1::basic_string_view<char, std::__1::char_traits<char> > > >::__insert_unique(std::__1::basic_string_view<char, std::__1::char_traits<char> >&&) /usr/local/bin/../include/c++/v1/__hash_table:1129:14
[2022-08-10T19:18:02.798Z]     #8 0x1adf3ad9 in std::__1::unordered_set<std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::hash<std::__1::basic_string_view<char, std::__1::char_traits<char> > >, std::__1::equal_to<std::__1::basic_string_view<char, std::__1::char_traits<char> > >, std::__1::allocator<std::__1::basic_string_view<char, std::__1::char_traits<char> > > >::insert(std::__1::basic_string_view<char, std::__1::char_traits<char> >&&) /usr/local/bin/../include/c++/v1/unordered_set:584:26
[2022-08-10T19:18:02.798Z]     #9 0x1adf3ad9 in DB::ExecutorIdGenerator::generate(tipb::Executor const&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Planner/ExecutorIdGenerator.h:38:13
[2022-08-10T19:18:02.798Z]     #10 0x1adef95e in DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1::operator()(tipb::Executor const&) const /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Planner/PhysicalPlan.cpp:70:32
[2022-08-10T19:18:02.798Z]     #11 0x1adef95e in void DB::traverseExecutorTreePostOrder<DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&>(tipb::Executor const&, DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Statistics/traverseExecutors.h:114:5
[2022-08-10T19:18:02.798Z]     #12 0x1adef95e in void DB::traverseExecutorTreePostOrder<DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&>(tipb::Executor const&, DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&)::'lambda'(tipb::Executor const&)::operator()(tipb::Executor const&) const /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Statistics/traverseExecutors.h:113:72
[2022-08-10T19:18:02.798Z]     #13 0x1adef611 in void DB::Children::forEach<void DB::traverseExecutorTreePostOrder<DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&>(tipb::Executor const&, DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&)::'lambda'(tipb::Executor const&)>(DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Statistics/traverseExecutors.h:47:13
[2022-08-10T19:18:02.798Z]     #14 0x1adef611 in void DB::traverseExecutorTreePostOrder<DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&>(tipb::Executor const&, DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Statistics/traverseExecutors.h:113:27
[2022-08-10T19:18:02.798Z]     #15 0x1adef611 in void DB::traverseExecutorTreePostOrder<DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1>(tipb::Executor const&, DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&&)::'lambda'(tipb::Executor const&)::operator()(tipb::Executor const&) const /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Statistics/traverseExecutors.h:113:72
[2022-08-10T19:18:02.798Z]     #16 0x1ade990a in void DB::Children::forEach<void DB::traverseExecutorTreePostOrder<DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1>(tipb::Executor const&, DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&&)::'lambda'(tipb::Executor const&)>(DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Statistics/traverseExecutors.h:49:17
[2022-08-10T19:18:02.798Z]     #17 0x1ade990a in void DB::traverseExecutorTreePostOrder<DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1>(tipb::Executor const&, DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Statistics/traverseExecutors.h:113:27
[2022-08-10T19:18:02.798Z]     #18 0x1ade990a in void DB::traverseExecutorsReverse<DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1>(tipb::DAGRequest const*, DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Statistics/traverseExecutors.h:129:9
[2022-08-10T19:18:02.798Z]     #19 0x1ade990a in DB::PhysicalPlan::build(tipb::DAGRequest const*) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Planner/PhysicalPlan.cpp:67:5
[2022-08-10T19:18:02.798Z]     #20 0x1ae0b0b4 in DB::Planner::executeImpl(DB::DAGPipeline&) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Planner/Planner.cpp:52:19
[2022-08-10T19:18:02.798Z]     #21 0x1ae0a82c in DB::Planner::execute() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Planner/Planner.cpp:36:5
[2022-08-10T19:18:02.798Z]     #22 0x19d7fa4a in DB::(anonymous namespace)::executeQueryImpl(DB::IQuerySource&, DB::Context&, bool, DB::QueryProcessingStage::Enum) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Interpreters/executeQuery.cpp:236:28
[2022-08-10T19:18:02.798Z]     #23 0x19d84bf0 in DB::executeQuery(DB::IQuerySource&, DB::Context&, bool, DB::QueryProcessingStage::Enum) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Interpreters/executeQuery.cpp:434:38
[2022-08-10T19:18:02.798Z]     #24 0x1aad5fbd in DB::executeQuery(DB::Context&, bool, DB::QueryProcessingStage::Enum) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/executeQuery.cpp:30:16
[2022-08-10T19:18:02.799Z]     #25 0x865edc6 in DB::tests::ExecutorTest::executeInterpreter(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::shared_ptr<tipb::DAGRequest> const&, unsigned long) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/TestUtils/ExecutorTestUtils.cpp:71:16
[2022-08-10T19:18:02.799Z]     #26 0x40a5cd1 in DB::tests::PlannerInterpreterExecuteTest_ParallelQuery_Test::TestBody() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/tests/gtest_planner_interpreter.cpp:463:9
[2022-08-10T19:18:02.799Z]     #27 0x14a6cff0 in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2443:10
[2022-08-10T19:18:02.799Z]     #28 0x14a6cff0 in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2479:14
[2022-08-10T19:18:02.799Z]     #29 0x14a0b798 in testing::Test::Run() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2517:5
[2022-08-10T19:18:02.799Z]     #30 0x14a0e927 in testing::TestInfo::Run() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2693:11
[2022-08-10T19:18:02.799Z]     #31 0x14a0fe9f in testing::TestCase::Run() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2811:28
[2022-08-10T19:18:02.799Z]     #32 0x14a3a8b8 in testing::internal::UnitTestImpl::RunAllTests() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:5177:43
[2022-08-10T19:18:02.799Z]     #33 0x14a6f5f0 in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2443:10
[2022-08-10T19:18:02.799Z]     #34 0x14a6f5f0 in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:2479:14
[2022-08-10T19:18:02.799Z]     #35 0x14a39131 in testing::UnitTest::Run() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/src/gtest.cc:4786:10
[2022-08-10T19:18:02.799Z]     #36 0x84e15aa in RUN_ALL_TESTS() /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/contrib/googletest/googletest/include/gtest/gtest.h:2341:46
[2022-08-10T19:18:02.799Z]     #37 0x84e15aa in main /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/TestUtils/gtests_dbms_main.cpp:77:16
[2022-08-10T19:18:02.799Z]     #38 0x7f88426bf554 in __libc_start_main (/lib64/libc.so.6+0x22554)
[2022-08-10T19:18:02.799Z]     #39 0x385f8a8 in _start (/tiflash/gtests_dbms+0x385f8a8)
[2022-08-10T19:18:02.799Z] 
[2022-08-10T19:18:02.799Z] Address 0x7fffff74a461 is located in stack of thread T0 at offset 33 in frame
[2022-08-10T19:18:02.799Z]     #0 0x1adef56f in void DB::traverseExecutorTreePostOrder<DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1>(tipb::Executor const&, DB::PhysicalPlan::build(tipb::DAGRequest const*)::$_1&&)::'lambda'(tipb::Executor const&)::operator()(tipb::Executor const&) const /home/jenkins/agent/workspace/tiflash-sanitizer-daily/tiflash/dbms/src/Flash/Statistics/traverseExecutors.h:113
[2022-08-10T19:18:02.799Z] 
[2022-08-10T19:18:02.799Z]   This frame has 1 object(s):
[2022-08-10T19:18:02.799Z]     [32, 56) 'ref.tmp.i.i' <== Memory access at offset 33 is inside this variable
[2022-08-10T19:18:02.799Z] HINT: this may be a false positive if your program uses some custom stack unwind mechanism, swapcontext or vfork
[2022-08-10T19:18:02.799Z]       (longjmp and C++ exceptions *are* supported)
[2022-08-10T19:18:02.799Z] SUMMARY: AddressSanitizer: stack-use-after-scope /llvm-project/compiler-rt/lib/asan/../sanitizer_common/sanitizer_common_interceptors.inc:860:7 in MemcmpInterceptorCommon(void*, int (*)(void const*, void const*, unsigned long), void const*, void const*, unsigned long)
[2022-08-10T19:18:02.799Z] Shadow bytes around the buggy address:
[2022-08-10T19:18:02.799Z]   0x10007fee1430: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
[2022-08-10T19:18:02.799Z]   0x10007fee1440: f1 f1 f1 f1 f8 f8 f2 f2 f8 f8 f2 f2 f8 f8 f8 f2
[2022-08-10T19:18:02.799Z]   0x10007fee1450: f2 f2 f2 f2 00 00 f3 f3 00 00 00 00 00 00 00 00
[2022-08-10T19:18:02.799Z]   0x10007fee1460: 00 00 00 00 00 00 00 00 00 00 00 00 f1 f1 f1 f1
[2022-08-10T19:18:02.799Z]   0x10007fee1470: 00 00 00 f3 f3 f3 f3 f3 00 00 00 00 00 00 00 00
[2022-08-10T19:18:02.799Z] =>0x10007fee1480: 00 00 00 00 00 00 00 00 f1 f1 f1 f1[f8]f8 f8 f3
[2022-08-10T19:18:02.799Z]   0x10007fee1490: f3 f3 f3 f3 00 00 00 00 00 00 00 00 00 00 00 00
[2022-08-10T19:18:02.799Z]   0x10007fee14a0: 00 00 00 00 f1 f1 f1 f1 f8 f8 f8 f2 f2 f2 f2 f2
[2022-08-10T19:18:02.799Z]   0x10007fee14b0: f8 f8 f8 f2 f2 f2 f2 f2 00 00 00 00 00 00 f2 f2
[2022-08-10T19:18:02.799Z]   0x10007fee14c0: f2 f2 00 00 f3 f3 f3 f3 00 00 00 00 00 00 00 00
[2022-08-10T19:18:02.799Z]   0x10007fee14d0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
[2022-08-10T19:18:02.799Z] Shadow byte legend (one shadow byte represents 8 application bytes):
[2022-08-10T19:18:02.799Z]   Addressable:           00
[2022-08-10T19:18:02.799Z]   Partially addressable: 01 02 03 04 05 06 07
[2022-08-10T19:18:02.799Z]   Heap left redzone:       fa
[2022-08-10T19:18:02.799Z]   Freed heap region:       fd
[2022-08-10T19:18:02.799Z]   Stack left redzone:      f1
[2022-08-10T19:18:02.799Z]   Stack mid redzone:       f2
[2022-08-10T19:18:02.799Z]   Stack right redzone:     f3
[2022-08-10T19:18:02.799Z]   Stack after return:      f5
[2022-08-10T19:18:02.799Z]   Stack use after scope:   f8
[2022-08-10T19:18:02.799Z]   Global redzone:          f9
[2022-08-10T19:18:02.800Z]   Global init order:       f6
[2022-08-10T19:18:02.800Z]   Poisoned by user:        f7
[2022-08-10T19:18:02.800Z]   Container overflow:      fc
[2022-08-10T19:18:02.800Z]   Array cookie:            ac
[2022-08-10T19:18:02.800Z]   Intra object redzone:    bb
[2022-08-10T19:18:02.800Z]   ASan internal:           fe
[2022-08-10T19:18:02.800Z]   Left alloca redzone:     ca
[2022-08-10T19:18:02.800Z]   Right alloca redzone:    cb
```

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: close #5600

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Planner/ExecutorIdGenerator.h [hunk 1]
```diff
@@ -78,6 +77,6 @@ class ExecutorIdGenerator
 
     UInt32 current_id = 0;
 
-    std::unordered_set<std::string_view> ids;
```
Comment (fuzhe1989) [inline dbms/src/Flash/Planner/ExecutorIdGenerator.h:81]: amazing ...

**Review by ywqzzy — APPROVED** [review]

File: dbms/src/Flash/Planner/ExecutorIdGenerator.h (hunk 1 above)
Comment (SeaRise) [reply dbms/src/Flash/Planner/ExecutorIdGenerator.h:81]: The lifetime of the string referenced by std::string_view only lives in `ExecutorIdGenerator::generate`.

**Review by Lloyd-Pottiger — APPROVED** [review]

**Pull request merged**